### ISSUE TITLE
Unable to skip granting permission to all files

### ISSUE THREAD

**Issue by vikdotdev** [ViscousPot/GitSync#42]
One of the initial setup screens that asks for all files permissions states that ["You can use GitSync without granting All files access ..."](https://github.com/ViscousPot/GitSync/blob/28c9b6ab52ebb66bf51ca0f07c138c3919452309/app/src/main/res/values/strings.xml#L56), which does not seem to be the case currently, as the prompt will re-appear repeatedly after the user refuses to grant the permission.

Some users won't be comfortable granting permissions to all files (including me). 

[Code for this](https://github.com/ViscousPot/GitSync/blob/28c9b6ab52ebb66bf51ca0f07c138c3919452309/app/src/main/java/com/viscouspot/gitsync/util/OnboardingController.kt#L176) is a bit over my head since I'm not an Android dev, but I don't see a code path where the user would be able to skip this step.

Basically description & code contradict - which one is correct?

Attaching the video of what I mean:
https://github.com/user-attachments/assets/fa96f47b-fa56-49b5-8621-28ce150556bc

As a side note - on my phone I somehow managed to hack around this dialog by restarting the app at some point, but can't reproduce it in the emulator.

**Comment by ViscousPot** [comment]
You are absolutely correct here. This is a remnant from past versions and I haven't yet updated the docs to reflect this either, so great catch!

GitSync does require "All files access" to function as it turns out. If I were to remove the persisting permission request, the syncing would seem to work, but would, in fact, not sync any newly created files or folders on the Android device. It's a little difficult to explain, but it's the reason that the Obsidian app also requires this permission (eventhough it only interacts with one folder) and is a limitation of Android

I'll go through and update the language to reflect this, but sadly users will have to give access to this permission. I'm hoping that, since the app is open-source, users can trust (or check) that the app is not attempting to touch any files outside the selected directory.

**Comment by ViscousPot** [comment]
Fixed in this [commit](https://github.com/ViscousPot/GitSync/commit/354b8ff7fe94cc180710251d046f346deef507af)

Will keep this issue open for some time to allow for discussion :)

**Comment by vikdotdev** [comment]
Thanks for the reply. 

Out of curiosity I tried investigating a bit deeper but can't wrap my head around why is it the case fundamentally that the app won't work with scoped permissions. [Obsidian docs on this](https://help.obsidian.md/Obsidian/Android+app#Storage+permissions) did not tell me much, but maybe it's because I don't know the Android APIs. [This Android docs page](https://developer.android.com/training/data-storage/manage-all-files) mentions "backup and restore apps" though.

Agree that app being open-source helps with trusting the app way-way more, but unfortunately it's not a perfect substitute for strict permissions. But oh well - the app use-case is pretty niche to become a serious attack vector, so most users will probably be fine with this.

**Comment by ViscousPot** [comment]
Totally agree with you here. Honestly the situation with Android storage permissions seems to be a mess. The older versions of GitSync used SAF (Storage Access Framework), but, since the implementation of Git I'm using doesn't support that api, I am forced to use File API compatible permissions and logic (which has an undocumented and limited compatibility with SAF). 

This is what is enforcing the "All files access" requirement. Without it, everything seems to work fine with SAF, but it silently does not sync any newly created files or directories.

Happy to hear any ways I can either make the risks more clear to users, or reduce the risk outright!

**Comment by ViscousPot** [comment]
Closing as resolved. :)
